### ISSUE TITLE
Segmentation Violation

### ISSUE THREAD

**Issue by Nimbus689** [hahwul/dalfox#289]
## Describe the bug
When running any type of scan dalfox throws a segmentation violation 

## Environment
 * Dalfox Version: 2.5.2
 * Installed from: go-get

![image](https://user-images.githubusercontent.com/89958947/135140354-87875ebb-749e-4b4d-9ea8-0e97442e4fd3.png)

**Comment by hahwul** [comment]
Hi @Nimbus689 
First, thank you for submit issue!

I checked the part in the code that could cause the error, although the error did not occur during the my test case. (If the server blocks or dies during the scan)

I added defense logic in dev branch. can you check?

## Test with dev branch
```
$ git clone https://github.com/hahwul/dalfox
$ git branch dev
$ git checkout dev
$ git pull -v
$ go build
$ ./dalfox
```

**Comment by Nimbus689** [comment]
I ran the dev branch code and it is still happening. It was working before I upgraded to 2.5.0.... I was able to check out 2.5.2 on a different machine and it runs as expected there must be something broken when I updated dalfox

![image](https://user-images.githubusercontent.com/89958947/135194815-74fe002a-7f47-4a7c-ad7f-34d18f09c429.png)

**Comment by hahwul** [comment]
@Nimbus689 
Yes, WAF detection and evasion have been added to `v2.5.0`, and it seems to be a problem that occurs in that area. 
It would be great if it could be reproduced in my environment, but I don't have any errors as below 😭

**Ubuntu**
![1111](https://user-images.githubusercontent.com/13212227/135200332-f258b6c1-3247-4ff8-b4d3-d8f598d85515.png)

**Kali (docker)**
![2222](https://user-images.githubusercontent.com/13212227/135200462-1c456f3b-6b68-4a68-bfaf-220d0d481f52.png)

Hm... I'll test it a bit more! It is a little strange that the problem occurs even though the write to exception.

**Comment by Nimbus689** [comment]
It's alright don't waste anymore time on it. I've been meaning to swap my machine over to a new one anyway so I'll just go ahead and do it :) 

Thanks for being responsive to the tickets and for making a great tool!

**Comment by panch0r3d** [comment]
I'm actually seeing this same error on a lot of URL's  after the upgrade. It's not happening on all URL's but I have automation that batch in a file of 100 at a time and it is throwing this error consistently before it finishes a batch. Would more URL's to test with help? 

[I] Found 15 testing point in DOM base parameter mining
 ⠼  [SA: ✓ PA: ✓ BAV: ✓ ] Waiting for analysis 🔍 panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0xa14eee]

goroutine 1745 [running]:
github.com/hahwul/dalfox/v2/pkg/scanning.ParameterAnalysis.func7(0xc000b1a000, 0xc000b7cb00, 0x7ffe64dcf3b1, 0x2c, 0xc000676040, 0xc001240020, 0xc001240018, 0xc000678020, 0xc000b460c0, 0xc001240010, ...)
	/home/panch0r3d/go/pkg/mod/github.com/hahwul/dalfox/v2@v2.5.2/pkg/scanning/parameterAnlaysis.go:202 +0x40e
created by github.com/hahwul/dalfox/v2/pkg/scanning.ParameterAnalysis
	/home/panch0r3d/go/pkg/mod/github.com/hahwul/dalfox/v2@v2.5.2/pkg/scanning/parameterAnlaysis.go:194 +0x3d3

**Comment by hahwul** [comment]
@panch0r3d cc) @Nimbus689 
Oh, I think that's a big problem. Is there any way, can you test the 2.5.3-dev version?

## 2.5.3-dev (dev branch)
```
$ git clone https://github.com/hahwul/dalfox
$ git branch dev
$ git checkout dev
$ git pull -v
$ go build
$ ./dalfox
```

**Comment by hahwul** [comment]
@Nimbus689 
[When i told you](https://github.com/hahwul/dalfox/issues/289#issuecomment-929684089) for setup dev branch command, the pull might not be right. Can you check again to see if it's possible with the command below?

```
git clone https://github.com/hahwul/dalfox
git branch dev
git checkout dev
git pull origin dev -v 
go build
./dalfox
```

**Comment by hahwul** [comment]
@panch0r3d @Nimbus689 
If it is fixed with dev branch code, I will proceed with hotfix release(v2.5.3) right away! 
(It's frustrating that I can't trigger problem in my machine😵)

**Comment by hahwul** [comment]
@panch0r3d @Nimbus689 
First off, I'll do hotfix process right away! If there's any issue after 2.5.3, please let me know!

I'm really sorry for causing you discomfort due to code problems! I wasn't very meticulous 😱 
Anyway, thank you so much for great help in solving this issue! Enjoy your week 🥰

**Comment by Nimbus689** [comment]
Hey I know it is closed already but I wanted to confirm that the issue has been resolved for me with the 2.5.3 release. So thanks again :D
